### PR TITLE
"Forwarding" Annotation calls to Variant

### PR DESCRIPTION
**Pull request by DavidPoliakoff**
This will allow us to do whatever we want in Variant, and have Annotation be able to play nicely with it. Good call by @mrzv that in the single-arg case we can do this without variadics, getting most of the benefit without C++11

The "std::enable_if" prevents infinite recursion, which I think would happen otherwise.

If this is good, we can put together another PR with changes to Variant including std::size_t and such

### CODE REVIEW

**Comment by DavidPoliakoff** [comment]
Quick note for @daboehme , I've only tested this with Clang/GCC/ICC. We might want to make sure other compilers don't choke on this

File: include/caliper/Annotation.h [hunk 1]
```diff
@@ -202,6 +207,39 @@ class Annotation
...
+        return begin(Variant(std::forward<Args>(args)...));
+    }
+    template<typename... Args>
+    auto set(Args&&... args) -> typename std::enable_if<!std::is_same<typename head<Args...>::type,cali::Variant>::value,Annotation&>::type {
+        return set(Variant(std::forward<Args>(args)...));
+    }
+#else
+    template<typename Arg>
+    Annotation& begin(const Arg& arg){
+        return begin(arg);
```
Comment (daboehme) [inline include/caliper/Annotation.h]: Don't we need `begin(Variant(arg))` here?

Comment (DavidPoliakoff) [reply include/caliper/Annotation.h]: Yiiiikes

Comment (DavidPoliakoff) [reply include/caliper/Annotation.h]: (Will fix)

**Comment by DavidPoliakoff** [comment]
Yep yep yep, no new changes pushed, definitely didn't screw up and forget the point of the PR on one conditional preprocessor branch

**Comment by daboehme** [comment]
Ok, nice. Tried it with PGI and xlC and it seems to work.

**Pull request merged**